src/inmanta/agent/agent.py

```diff
@@ -1526,3 +1527,149 @@ async def get_facts(self, env: uuid.UUID, agent: str, resource: dict[str, Any])
     @protocol.handle(methods.get_status)
     async def get_status(self) -> Apireturn:
         return 200, collect_report(self)
+
+
+@dataclasses.dataclass(frozen=True)
+class EnvBlueprint:
+    pip_config: PipConfig
+    requirements: Sequence[str]
+
+
+@dataclasses.dataclass(frozen=True)
+class ExecutorBlueprint(EnvBlueprint):
+    sources: Sequence[ModuleSource]
+
+
+@dataclasses.dataclass(frozen=True)
+class ExecutorId:
+    agent_name: str
+    blueprint: ExecutorBlueprint
+
+
+class ExecutorVirtualEnvironment:
+    def __init__(self, storage: str, threadpool: ThreadPoolExecutor):
+        self.env = env.VirtualEnv(storage)
```

Comment:
I think you should use 'PythonEnvironment' 

This one is a child of `ActiveEnv` which it clearly is not. 

I would see how difficult it is to lift the required functionality from `VirtualEnv` to `ActiveEnv`